lib/acts_as_descriptable.rb

```diff
@@ -46,20 +43,13 @@ def descriptor_value(key)
 
   def add_descriptor(descriptor)
     write_attribute(:descriptors, descriptor_hash.merge(descriptor.name => descriptor.value))
-    write_attribute(:descriptor_fields, descriptor_fields.push(descriptor.name))
   end
 
   def descriptor_hash
     read_attribute(:descriptors) || {}
   end
 
-  private
-
-  def each_descriptor
-    descriptor_fields.each do |field|
-      next if field.blank?
-
-      yield(field, descriptor_hash[field])
-    end
+  def descriptor_fields=(fields)
+    # Ignore for now while we remove this
```

Comment:
Maybe a slightly fuller explanation? Could see this persisting into the future due to confusion...

Reply:
Its already gone in the head of my branch.